### PR TITLE
Test block context methods

### PR DESCRIPTION
**Pull request by Iwueseiter**

### CODE REVIEW

File: concrete/api/environment_test.go
Comment (therealbytes) [inline concrete/api/environment_test.go:1]: This is in the right direction but the tests need to be more exhaustive. I just merged a fix to main that includes an example test to get you started:

```go
func TestBlockContextMethods(t *testing.T) {
	var (
		r        = require.New(t)
		config   = EnvConfig{IsStatic: true, IsTrusted: false}
		meterGas = true
		gas      = uint64(1e6)
	)

	env, _, _block, _ := NewMockEnvironment(config, meterGas)
	block := _block.(*mockBlockContext)

	t.Run("BlockHash", func(t *testing.T) {
		env.contract.Gas = gas

		var (
			currentBlockNumber = uint64(400)
			lastBlockNumber    = currentBlockNumber - 1
			recentBlockNumber  = currentBlockNumber - 2
			limitBlockNumber   = lastBlockNumber - 255
			limitBlockNumberS1 = limitBlockNumber - 1
			futureBlockNumber  = currentBlockNumber + 1
		)

		block.SetBlockNumber(currentBlockNumber)
		block.SetBlockHash(currentBlockNumber, common.Hash{0x01})
		block.SetBlockHash(lastBlockNumber, common.Hash{0x02})
		block.SetBlockHash(recentBlockNumber, common.Hash{0x03})
		block.SetBlockHash(limitBlockNumber, common.Hash{0x04})
		block.SetBlockHash(limitBlockNumberS1, common.Hash{0x05})
		block.SetBlockHash(futureBlockNumber, common.Hash{0x06})

		r.Equal(common.Hash{}, env.GetBlockHash(currentBlockNumber))
		r.Equal(block.GetHash(lastBlockNumber), env.GetBlockHash(lastBlockNumber))
		r.Equal(block.GetHash(recentBlockNumber), env.GetBlockHash(recentBlockNumber))
		r.Equal(block.GetHash(limitBlockNumber), env.GetBlockHash(limitBlockNumber))
		r.Equal(common.Hash{}, env.GetBlockHash(limitBlockNumberS1))
		r.Equal(common.Hash{}, env.GetBlockHash(futureBlockNumber))

		r.Equal(env.Gas(), gas-6*GasExtStep)
	})

	t.Run("GasLimit", func(t *testing.T) {
		// ...
	})

	// ...

}
```

**Review by therealbytes — CHANGES_REQUESTED** [review]
This is in the right direction but the tests need to be more exhaustive. I just merged a fix to main that includes an example test to get you started:

```go
func TestBlockContextMethods(t *testing.T) {
	var (
		r        = require.New(t)
		config   = EnvConfig{IsStatic: true, IsTrusted: false}
		meterGas = true
		gas      = uint64(1e6)
	)

	env, _, _block, _ := NewMockEnvironment(config, meterGas)
	block := _block.(*mockBlockContext)

	t.Run("BlockHash", func(t *testing.T) {
		env.contract.Gas = gas

		var (
			currentBlockNumber = uint64(400)
			lastBlockNumber    = currentBlockNumber - 1
			recentBlockNumber  = currentBlockNumber - 2
			limitBlockNumber   = lastBlockNumber - 255
			limitBlockNumberS1 = limitBlockNumber - 1
			futureBlockNumber  = currentBlockNumber + 1
		)

		block.SetBlockNumber(currentBlockNumber)
		block.SetBlockHash(currentBlockNumber, common.Hash{0x01})
		block.SetBlockHash(lastBlockNumber, common.Hash{0x02})
		block.SetBlockHash(recentBlockNumber, common.Hash{0x03})
		block.SetBlockHash(limitBlockNumber, common.Hash{0x04})
		block.SetBlockHash(limitBlockNumberS1, common.Hash{0x05})
		block.SetBlockHash(futureBlockNumber, common.Hash{0x06})

		r.Equal(common.Hash{}, env.GetBlockHash(currentBlockNumber))
		r.Equal(block.GetHash(lastBlockNumber), env.GetBlockHash(lastBlockNumber))
		r.Equal(block.GetHash(recentBlockNumber), env.GetBlockHash(recentBlockNumber))
		r.Equal(block.GetHash(limitBlockNumber), env.GetBlockHash(limitBlockNumber))
		r.Equal(common.Hash{}, env.GetBlockHash(limitBlockNumberS1))
		r.Equal(common.Hash{}, env.GetBlockHash(futureBlockNumber))

		r.Equal(env.Gas(), gas-6*GasExtStep)
	})

	t.Run("GasLimit", func(t *testing.T) {
		// ...
	})

	// ...

}
```

**Comment by therealbytes** [comment]
Note you will need to rebase on main for your tests to pass.

**Comment by Iwueseiter** [comment]
@therealbytes I will work on it.

**Comment by therealbytes** [comment]
Hey, any progress here?

**Review by therealbytes — CHANGES_REQUESTED** [review]
(edit: commented in the wrong issue)

**Comment by Iwueseiter** [comment]
@therealbytes I'm working on it. I will update the PR today.

**Review by therealbytes — CHANGES_REQUESTED** [review]
Looking good. I think you accidentally deleted `TestCallMethods` while resolving merge conflicts.
- `TestBlockOps_Minimal` can be deleted
- `TestCallMethods` should NOT be deleted

**Comment by Iwueseiter** [comment]
@therealbytes I will rectify that.

File: concrete/api/environment_test.go [hunk 1]
```diff
@@ -15,9 +15,8 @@
 
 //go:build !tinygo
 
-// This file will ignored when building with tinygo to prevent compatibility
+// This file will be ignored when building with tinygo to prevent compatibility
 // issues.
-
 package api
```
Comment (therealbytes) [inline concrete/api/environment_test.go:21]: Needs an empty line before this line (under `// issues.`),

File: concrete/api/environment_test.go [hunk 2]
```diff
@@ -185,28 +205,76 @@ func TestBlockContextMethods(t *testing.T) {
...
+		env.contract.Gas = gas
+
+		blockRandom := common.Hash{0x01}
+		block.SetRandom(blockRandom)
+
+		r.Equal(blockRandom, env.GetPrevRandom())
+		r.Equal(env.Gas(), gas-GasQuickStep)
+	})
+}
 func TestCallMethods(t *testing.T) {
```
Comment (therealbytes) [inline concrete/api/environment_test.go:280]: Needs an empty line before this line (under `}`).

**Review by therealbytes — CHANGES_REQUESTED** [review]

**Comment by Iwueseiter** [comment]
@therealbytes  Done

**Review by therealbytes — APPROVED** [review]

**Pull request merged**